File: ljh/javascript/level1/GCDandLCM.js
export default class GCDandLCM {
    solution = (n, m) => {
        let large = n;
        let small = m;
        let answer = [];

        if(n<m){
            large = m;
            small = n;
        }

        let temp = large % small;

        while(temp != 0){
            large = small;
            small = temp;
            temp = large % small;
        }

        answer.push(small);
        answer.push(n*m/small);
        
        return answer;
    }
}

File: Ywy/Go/level1/최대공약수와최소공배수/gcdAndLcm.go
package gcdAndLcm

func Solution(n, m int) (answer []int) {
	gcd := Gcd(n, m)
	lcm := m / gcd * n
	return []int{gcd, lcm}
}

func Gcd(n, m int) int {
	if m == 0 {
		return n
	}
	return Gcd(m, n%m)
}
